core/testcontainers/core/docker_client.py

```diff
@@ -102,11 +103,14 @@ def host(self) -> str:
 
         except ValueError:
             return None
+        adapter = self.client.api.get_adapter(self.client.api.base_url)
+        is_ipc = isinstance(adapter, UnixHTTPAdapter)
+        is_ipc |= hasattr(adapter, "socket_path") or hasattr(adapter, "npipe_path")
+        is_ipc |= 'unix' in url.scheme or 'npipe' in url.scheme
```

Comment:
Even if the `NpipeHTTPAdapter` is only used on windows, would it still be available for an `isinstance` check independent of the platform? Or does the `docker` package not even expose `NpipeHTTPAdapter` if it's not on windows?

Alternatively, would checking `adapter.__class__.name == 'NpipeHTTPAdapter'` be easier to understand than checking the attributes of the adapter?

Reply:
I don't think that `NpipeHTTPAdapter` symbol is available outside of windows. Checking the class name would break for `NpipeHTTPAdapter` subclasses. So I went on with a classic duck-typing.

Reply:
I think something like this:

```python
try:
    from docker.transport.npipeconn import NpipeHTTPAdapter
except ImportError:
    class NpipeHTTPAdapter: ...
```

Would be a more proper solution.

It is much clearer what happens, does not depend the design of the Adapter and respects subclasses.